transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -1053,4 +1079,53 @@ boolean setClosed() {
             return super.trySuccess();
         }
     }
+
+    private static final class AutoFlushTask implements Runnable {
+
+        private final AbstractChannel channel;
+        private final SingleThreadEventLoop eventLoop;
+        private boolean writePending;
+        private boolean running;
+
+        AutoFlushTask(AbstractChannel channel) {
+            this.channel = channel;
+            assert channel.eventLoop() instanceof SingleThreadEventLoop;
+            eventLoop = (SingleThreadEventLoop) channel.eventLoop();
+        }
+
+        @Override
+        public void run() {
+            running = true;
+            try {
+                if (writePending) {
+                    writePending = false;
+                    channel.flush();
+                    if (writePending) {
```

Comment:
@NiteshKant I can not see how this could ever be true again here... We only update this to true from the run() method which can not run concurrently... Can you explain this ?